server/src/main/java/org/elasticsearch/snapshots/InternalSnapshotsInfoService.java

```diff
@@ -120,7 +120,7 @@ private void setMaxConcurrentFetches(Integer maxConcurrentFetches) {
     @Override
     public SnapshotShardSizeInfo snapshotShardSizes() {
         synchronized (mutex){
-            final ImmutableOpenMap.Builder<SnapshotShard, Long> snapshotShardSizes = ImmutableOpenMap.builder(knownSnapshotShardSizes);
+            final ImmutableOpenMap.Builder<SnapshotShard, Long> snapshotShardSizes = ImmutableOpenMap.builder(knownSnapshotShards);
             for (SnapshotShard snapshotShard : failedSnapshotShards) {
```

Comment:
I think we could change the implementation to only have one map now, by putting the `UNAVAILABLE_EXPECTED_SHARD_SIZE` value into the map when fetching fails. But we can leave that to follow-ups, we may want to refine the failure handling and that could benefit from having them in separate maps.

Reply:
Having two maps makes it easier (at least for me) to reason about how failures are handled, so I'd prefer to keep it like this for now if that's OK for you.

Reply:
That is fine, but could we then check whether `failedSnapshotShards.isEmpty()` and return `knownSnapshotShards` directly if it is? Just to not waste cpu cycles unnecessarily on the happy case.

Reply:
Makes sense.